common_types/src/record_batch.rs

```diff
@@ -269,19 +269,30 @@ impl RecordBatch {
     }
 }
 
-impl TryFrom<ArrowRecordBatch> for RecordBatch {
-    type Error = Error;
+#[derive(Default)]
+pub struct RecordBatchBuilder {
+    schema_cache: Option<RecordSchema>,
+}
 
-    fn try_from(arrow_record_batch: ArrowRecordBatch) -> Result<Self> {
-        let record_schema =
-            RecordSchema::try_from(arrow_record_batch.schema()).context(ConvertArrowSchema)?;
+impl RecordBatchBuilder {
+    pub fn build(&mut self, arrow_record_batch: ArrowRecordBatch) -> Result<RecordBatch> {
```

Comment:
It's wired to name this `build`, usually build is only called once.

`convert` maybe a better name.

Reply:
Done.